pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -152,6 +204,43 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 		}
 	}
 
+	ann := instance.GetAnnotations()
+	if _, ok := ann[cmpv1alpha1.DisableOutdatedReferenceValidation]; ok {
+		reqLogger.Info("Reference validation is disabled, skipping validation")
+	} else if isValidationRequired(instance) {
+		reqLogger.Info("Validating TailoredProfile")
+		pruneOutdated := false
+		if _, ok := ann[cmpv1alpha1.PruneOutdatedReferencesAnnotationKey]; ok {
```

Comment:
This only checks if the annotation exists, but in our tests we're setting this to `true` at times. Which should we be using?

Setting it to `true` feels more explicit to me, with the absence or `false` value being the default.

Reply:
I think this makes sense here